packages/react-checkbox/src/components/Checkbox/useCheckboxStyles.ts

```diff
@@ -1,26 +1,244 @@
+    },
+  }),
+
+  checked: theme => ({
+    backgroundColor: theme.alias.color.neutral.compoundBrandBackground,
+    borderWidth: 0,
+    ':hover': {
+      backgroundColor: theme.alias.color.neutral.compoundBrandBackgroundHover,
+    },
+    ':active': {
+      backgroundColor: theme.alias.color.neutral.compoundBrandBackgroundPressed,
+    },
+  }),
+
+  mixed: theme => ({
+    borderColor: theme.alias.color.neutral.compoundBrandStroke,
+    borderWidth: theme.global.strokeWidth.thin,
+    ':hover': {
+      borderColor: theme.alias.color.neutral.compoundBrandStrokeHover,
+    },
+    ':active': {
+      borderColor: theme.alias.color.neutral.compoundBrandStrokePressed,
+    },
+  }),
+});
+
+const useIndicatorStyles = makeStyles({
+  indicator: {
+    position: 'absolute',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    fill: 'currentColor',
+  },
+
+  // TODO: Remove fontSize once checkbox uses react-icons
+  medium: {
+    fontSize: '8px',
+    width: '8px',
+    height: '8px',
+  },
+
+  // TODO: Remove fontSize once checkbox uses react-icons
+  large: {
+    fontSize: '10px',
+    width: '10px',
+    height: '10px',
+  },
+
+  disabled: theme => ({
+    opacity: 1,
+    color: theme.alias.color.neutral.neutralForegroundDisabled,
+  }),
+
+  unchecked: {
+    opacity: 0,
+  },
+
+  // TODO: neutralForegroundInverted change to NeutralForegroundOnBrand once it's added
+  checked: theme => ({
+    opacity: 1,
+    color: theme.alias.color.neutral.neutralForegroundInverted,
+  }),
+
+  mixed: theme => ({
+    opacity: 1,
+    color: theme.alias.color.neutral.compoundBrandForeground1,
+    ':hover': {
+      color: theme.alias.color.neutral.compoundBrandForeground1Hover,
+    },
+    ':active': {
+      color: theme.alias.color.neutral.compoundBrandForeground1Pressed,
+    },
+  }),
+});
+
+const useLabelStyles = makeStyles({
+  label: {
+    userSelect: 'none',
+    cursor: 'pointer',
+  },
+
+  disabled: theme => ({
+    color: theme.alias.color.neutral.neutralForegroundDisabled,
+    cursor: 'default',
+  }),
+
+  unchecked: theme => ({
+    color: theme.alias.color.neutral.neutralForeground3,
+    ':hover': {
+      color: theme.alias.color.neutral.neutralForeground2,
+    },
+    ':active': {
+      color: theme.alias.color.neutral.neutralForeground1,
+    },
+  }),
+
+  checked: theme => ({
+    color: theme.alias.color.neutral.neutralForeground1,
+  }),
+
+  mixed: theme => ({
+    color: theme.alias.color.neutral.neutralForeground1,
+  }),
 });
 
 /**
  * Apply styling to the Checkbox slots based on the state
  */
 export const useCheckboxStyles = (state: CheckboxState): CheckboxState => {
+  const checkedState = state.checked === 'mixed' ? 'mixed' : state.checked ? 'checked' : 'unchecked';
+  const indicatorStyles = useIndicatorStyles();
+  const labelStyles = useLabelStyles();
+  const inputStyles = useInputStyle();
+  const boxStyles = useBoxStyles();
```

Comment:
_Just for curiosity_: what is a reason to have multiple `makeStyles()` call?

```tsx
const styles = makeStyles({
  root: {},
  label: {},
  inputChecked: {},
})

// vs

const styles = makeStyles({ root: {} })
const labelStyles = makeStyles({ root: {} })
const inputStyles = makeStyles({ checked: {} })
```

Reply:
It is mainly for cleanliness, but am working on refactoring the styles so this isn't final.